tests/phpunit/tests/test-admin-filters-post.php

```diff
@@ -437,18 +437,20 @@ function test_categories_script_data_in_footer() {
 		set_current_screen( 'edit' );
 		$GLOBALS['wp_scripts'] = new WP_Scripts();
 		wp_default_scripts( $GLOBALS['wp_scripts'] );
+
+		$term_id = $this->factory->term->create( array( 'taxonomy' => 'category', 'name' => 'test' ) );
+		self::$model->term->set_language( $term_id, 'fr' );
+
 		do_action( 'admin_enqueue_scripts' );
 
 		ob_start();
 		do_action( 'admin_print_footer_scripts' );
 		$footer = ob_get_clean();
 
-		// All categories we have i.e. default categories
-		foreach ( self::$model->get_languages_list() as $lang ) {
-			$terms[ $lang->slug ] = array( 'category' => array( self::$model->term->get( 1, $lang->slug ) ) );
-		}
+		$terms['fr'] = array( 'category' => array( self::$model->term->get( $term_id, 'fr' ) ) );
 
-		$this->assertNotFalse( strpos( $footer, 'var pll_term_languages = ' . wp_json_encode( $terms ) ) );
+		// Deletion of the last three characters of the string we are looking for, because we also have the default category ( {"fr":{"category":[term_fr]}} )
+		$this->assertNotFalse( strpos( $footer, substr( 'var pll_term_languages = ' . wp_json_encode( $terms ), 0, -3 ) ) );
```

Comment:
Maybe rewrite what we test in the assertion. It isn't clear now we have to test all the javascript variable and it strange to me to truncate the returned value.